lizrd/core/llm.py

```diff
@@ -286,13 +322,32 @@ def PreNormBlock(dmodel, layer, name):
     )
 
 
-@ash.check("... d -> ... d")
-def TransformerBlock(dmodel, layers, gradient_checkpointing, residual_fn):
-    residual_fn = default(residual_fn, partial(PreNormBlock, dmodel=dmodel))
-    residual_layers = [residual_fn(layer=layer, name=name) for name, layer in layers]
-    if gradient_checkpointing:
-        residual_layers = [Checkpoint(layer) for layer in residual_layers]
-    return nn.Sequential(*residual_layers)
+class TransformerBlock(nn.Sequential):
+    def __init__(
+        self,
+        dmodel,
+        layers,
+        gradient_checkpointing,
+        residual_fn,
+        wrap_attn_and_ff_in_fsdp=False,
+        rank=None,
+        param_precision=torch.float32,
```

Comment:
this is non-descriptive. I assume this is the precision we optionally cast to?
I propose `selective_precision_dtype`

Reply:
added correct prefix